archethic/contracts/token_pool.exs

```diff
@@ -130,6 +130,25 @@ actions triggered_by: transaction, on: reveal_secret(htlc_genesis_address) do
   Contract.add_recipient address: htlc_genesis_address, action: "reveal_secret", args: [secret, signature]
 end
 
+condition triggered_by: transaction, on: update_code(new_code), as: [
+  previous_public_key: (
+		# Pool code can only be updated from the master chain if the bridge
+
+		# Transaction is not yet validated so we need to use previous address
+		# to get the genesis address
+		previous_address = Chain.get_previous_address()
+		Chain.get_genesis_address(previous_address) == #MASTER_GENESIS_ADDRESS#
+	),
+	code: Code.is_valid?(new_code)
```

Comment:
I would prefer to host the code in the content part of the contract instead of the function argument.
This is the purpose of the content's section of the transaction.

Probably function/action arguments should be limited, probably in terms of sizing.
Others protocols use function variable because they don't have a free zone for arbitrary data.
Since its inception, this was the goal of the `content` sector.

Also in terms of readability and auditing, this will also easier to read the code from the content zone than scanning the function's arguments...